templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -54,14 +57,23 @@ def generate_datasource():
     ingestion_pipeline.vector_store = vector_store
 
     # Run the ingestion pipeline and store the results
-    ingestion_pipeline.run(show_progress=True, documents=documents)
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
 
     # Default vector store only keeps data in memory, so we need to persist it
     # Can remove if using a different vector store
     if isinstance(vector_store, SimpleVectorStore):
-        vector_store.persist(os.path.join(STORAGE_DIR, "vector_store.json"))
-    # Persist the docstore to apply ingestion strategy
-    docstore.persist(os.path.join(STORAGE_DIR, "docstore.json"))
+        index = VectorStoreIndex(
+            nodes=nodes,
+            storage_context=StorageContext.from_defaults(
```

Comment:
`StorageContext` has a `persist` function that persists vec, index and doc store, can't we use it instead?

Reply:
Just answered above, please check: https://github.com/run-llama/create-llama/pull/71#discussion_r1587025184

Reply:
Ah, i thought you said applying `storage_context.persist` for other vector stores below. But this condition is already persited by `index.storagecontext.persist(STORAGE_DIR)`. which is the same as your idea right?